mmv1/third_party/terraform/resources/resource_dns_record_set.go

```diff
@@ -0,0 +1,389 @@
+		}
+		domain := mz.DnsName
+
+		if domain == d.Get("name").(string) {
+			log.Println("[DEBUG] NS records can't be deleted due to API restrictions, so they're being left in place. See https://www.terraform.io/docs/providers/google/r/dns_record_set.html for more information.")
+			return nil
+		}
+	}
+
+	// Build the change
+	chg := &dns.Change{
+		Deletions: []*dns.ResourceRecordSet{
+			{
+				Name:    d.Get("name").(string),
+				Type:    d.Get("type").(string),
+				Ttl:     int64(d.Get("ttl").(int)),
+				Rrdatas: rrdata(d),
+			},
+		},
+	}
+
+	log.Printf("[DEBUG] DNS Record delete request: %#v", chg)
+	chg, err = config.NewDnsClient(userAgent).Changes.Create(project, zone, chg).Do()
+	if err != nil {
+		return handleNotFoundError(err, d, "google_dns_record_set")
+	}
+
+	w := &DnsChangeWaiter{
+		Service:     config.NewDnsClient(userAgent),
+		Change:      chg,
+		Project:     project,
+		ManagedZone: zone,
+	}
+	_, err = w.Conf().WaitForState()
+	if err != nil {
+		return fmt.Errorf("Error waiting for Google DNS change: %s", err)
+	}
+
+	d.SetId("")
+	return nil
+}
+
+func resourceDnsRecordSetUpdate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	userAgent, err := generateUserAgentString(d, config.userAgent)
+	if err != nil {
+		return err
+	}
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	zone := d.Get("managed_zone").(string)
+	recordName := d.Get("name").(string)
+
+	oldTtl, newTtl := d.GetChange("ttl")
+	oldType, newType := d.GetChange("type")
+
+	oldCountRaw, _ := d.GetChange("rrdatas.#")
+	oldCount := oldCountRaw.(int)
+
+	chg := &dns.Change{
+		Deletions: []*dns.ResourceRecordSet{
+			{
+				Name:    recordName,
+				Type:    oldType.(string),
+				Ttl:     int64(oldTtl.(int)),
+				Rrdatas: make([]string, oldCount),
+			},
+		},
+		Additions: []*dns.ResourceRecordSet{
+			{
+				Name:    recordName,
+				Type:    newType.(string),
+				Ttl:     int64(newTtl.(int)),
+				Rrdatas: rrdata(d),
+			},
+		},
+	}
+
+	for i := 0; i < oldCount; i++ {
+		rrKey := fmt.Sprintf("rrdatas.%d", i)
+		oldRR, _ := d.GetChange(rrKey)
+		chg.Deletions[0].Rrdatas[i] = oldRR.(string)
+	}
+	log.Printf("[DEBUG] DNS Record change request: %#v old: %#v new: %#v", chg, chg.Deletions[0], chg.Additions[0])
+	chg, err = config.NewDnsClient(userAgent).Changes.Create(project, zone, chg).Do()
+	if err != nil {
+		return fmt.Errorf("Error changing DNS RecordSet: %s", err)
+	}
+
+	w := &DnsChangeWaiter{
+		Service:     config.NewDnsClient(userAgent),
+		Change:      chg,
+		Project:     project,
+		ManagedZone: zone,
+	}
+	if _, err = w.Conf().WaitForState(); err != nil {
+		return fmt.Errorf("Error waiting for Google DNS change: %s", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s/%s", zone, recordName, newType))
+
+	return resourceDnsRecordSetRead(d, meta)
+}
+
+func resourceDnsRecordSetImportState(d *schema.ResourceData, _ interface{}) ([]*schema.ResourceData, error) {
```

Comment:
Can we add support for `projects/(?P<project>[^/]+)/managedZones/(?P<managed_zone>[^/]+)/rrsets/(?P<name>[^/]+)/(?P<type>[^/]+)` as well? (also remember to update the docs- handwritten resources 🙁 )

Reply:
Added